### PR TITLE
Feature/pelagos 4437 add message and message handler for doi

### PR DESCRIPTION
**Pull request by praneethpr**

### CODE REVIEW

File: config/packages/old_sound_rabbit_mq.yaml [hunk 1]
```diff
@@ -2,27 +2,27 @@ old_sound_rabbit_mq:
...
+#        doi_issue:
+#            connection:       default
+#            exchange_options: { name: 'doi_issue', type: topic }
+#            callback:         App\Consumer\DoiConsumer
+#            qos_options:      { prefetch_count: 1, global: false }
+#            queue_options:
+#                name: 'doi_issue'
+#                routing_keys:
+#                    - 'doi'
+#                    - 'delete'
```
Comment (mickel1138) [inline config/packages/old_sound_rabbit_mq.yaml:28]: Ok to remove.

File: src/Command/RabbitUpdateAllAcceptedDatasetDOICommand.php [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 namespace App\Command;
 
+use App\Message\DoiMessage;
```
Comment (mickel1138) [inline src/Command/RabbitUpdateAllAcceptedDatasetDOICommand.php:5]: Also rename command to not mention Rabbit.

File: src/Message/DoiMessage.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+     */
+    private $action;
+
+    /**
+     * DoiMessage constructor.
+     *
+     * @param string $messageId Id of the dataset or doi identifier.
+     * @param string $action    Action that needs to be taken on the Doi Message.
+     */
+    public function __construct(string $messageId, string $action)
```
Comment (mickel1138) [inline src/Message/DoiMessage.php:41]: Maybe change messageId to context. And even make it second param them.

File: src/MessageHandler/DoiMessageHandler.php [hunk 4]
```diff
@@ -57,83 +37,66 @@ class DoiConsumer implements ConsumerInterface
...
      *
-     * @param AMQPMessage $message A filer message.
-     *
-     * @return integer
+     * @param DoiMessage $doiMessage The Doi Message that has to be handled.
      */
-    public function execute(AMQPMessage $message)
+    public function __invoke(DoiMessage $doiMessage)
     {
         $this->logger->info('Request received, waiting...');
```
Comment (mickel1138) [inline src/MessageHandler/DoiMessageHandler.php:64]: Not needed now, no sleep. Change message a bit.

File: src/Command/RabbitUpdateAllAcceptedDatasetDOICommand.php (hunk 2 above)
Comment (praneethpr) [reply src/Command/RabbitUpdateAllAcceptedDatasetDOICommand.php:5]: Renamed it

File: config/packages/messenger.yaml [hunk 5]
```diff
@@ -6,7 +6,7 @@ framework:
         transports:
             # https://symfony.com/doc/current/messenger.html#transport-configuration
             async: '%env(MESSENGER_TRANSPORT_DSN)%'
-            # failed: 'doctrine://default?queue_name=failed'
+            failed: 'doctrine://default?queue_name=failed'
```
Comment (mickel1138) [inline config/packages/messenger.yaml:9]: Should we use the same ENV variable form above, but add the queue name?

File: src/Event/DoctrineDatasetListener.php [hunk 6]
```diff
@@ -103,7 +104,8 @@ public function postUpdate(Dataset $dataset, LifecycleEventArgs $args): void
                 ) or
                 ($entity->getDif() instanceof DIF and $entity->getIdentifiedStatus() === DIF::STATUS_APPROVED)
             ) {
-                $this->publisher->publish($dataset->getId(), RabbitPublisher::DOI_PRODUCER, 'doi');
+                $doiMessage = new DoiMessage($entity->getId(), DoiMessage::ISSUE_OR_UPDATE);
```
Comment (mickel1138) [inline src/Event/DoctrineDatasetListener.php:107]: Unsure what $dataset is and $entity is (if the Entity is Dataset).

File: config/packages/messenger.yaml [hunk 7]
```diff
@@ -6,7 +6,7 @@ framework:
         transports:
```
Comment (mickel1138) [inline config/packages/messenger.yaml:6]: I think this `failure_transport` need to be uncommented.

File: src/Event/DoctrineDatasetListener.php (hunk 6 above)
Comment (praneethpr) [reply src/Event/DoctrineDatasetListener.php:107]: In this scenario, $entity and $dataset should be Dataset

File: src/Command/HandleSubmissionWithBadGMLCommand.php [hunk 8]
```diff
@@ -103,16 +92,8 @@ protected function execute(InputInterface $input, OutputInterface $output)
             }
         }
 
-        $this->entityManager->persist($datasetSubmission);
         $this->entityManager->flush();
 
-        //re-trigger dataset submission producer
-        $this->publisher->publish(
```
Comment (mickel1138) [inline src/Command/HandleSubmissionWithBadGMLCommand.php:110]: Maybe instead update the status.

**Review by mickel1138 — APPROVED** [review]
Tested, reviewed. LGTM!

**Review by fightingtexasaggie — APPROVED** [review]
reviewed in group.

**Pull request merged**